pkg/platform/kube/resourcescaler/resourcescaler.go

```diff
@@ -66,25 +70,43 @@ func New(kubeconfigPath string, namespace string) (scaler_types.ResourceScaler,
 
 func (n *NuclioResourceScaler) SetScale(resource scaler_types.Resource, scale int) error {
 	if scale == 0 {
-		return n.scaleFunctionToZero(n.namespace, string(resource))
+		return n.scaleFunctionToZero(n.namespace, resource.Name)
 	}
-	return n.scaleFunctionFromZero(n.namespace, string(resource))
+	return n.scaleFunctionFromZero(n.namespace, resource.Name)
 }
 
 func (n *NuclioResourceScaler) GetResources() ([]scaler_types.Resource, error) {
+	var functionList []scaler_types.Resource
 	functions, err := n.nuclioClientSet.NuclioV1beta1().NuclioFunctions(n.namespace).List(metav1.ListOptions{})
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to list functions")
 	}
 
-	var functionList []scaler_types.Resource
-
 	// build a list of function names that are potential to be scaled to zero
 	for _, function := range functions.Items {
 
 		// don't include functions that aren't in ready state or that min replicas is larger than zero
 		if function.GetComputedMinReplicas() <= 0 && function.Status.State == functionconfig.FunctionStateReady {
-			functionList = append(functionList, scaler_types.Resource(function.Name))
+			if function.Spec.ScaleToZero == nil {
+				return nil, errors.New("Function missing scale to zero spec")
```

Comment:
warn and go to the next function, otherwise 1 function with malformed spec (minreplicas=0 but no scz spec) will prevent scaling to zero of all functions